### PR TITLE
lvm2: support static compilation

### PR DESCRIPTION
**Pull request by baloo**
###### Motivation for this change

Fixup compilation of lvm2 on musl.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/lvm2/no-dynamic.patch [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
++                          libraries [default=yes]
+   --disable-thin_check_needs_check
+                           required if thin_check version is < 0.3.0
+   --disable-cache_check_needs_check
+@@ -7842,6 +7858,8 @@ main ()
+     if (*(data + i) != *(data3 + i))
+       return 14;
+   close (fd);
++  free (data);
++  free (data3);
```
Comment (7c6f434c) [inline pkgs/os-specific/linux/lvm2/no-dynamic.patch:90]: Does this mean there was a memory leak upstream? Not sure I like this being silently in the middle of allow-static-build patch…

Comment (baloo) [reply pkgs/os-specific/linux/lvm2/no-dynamic.patch:90]: `configure` (the output from autoconf) is commited in the repository. I ran autoconf, with a slightly more recent version than last maintainer of that file did (more than a year ago).
It looks like it's code that tests features from the environment to detect them, as autoconf usually does. This looks like fixes further upstream in autoconf.

Just like the `runstate` parameter additions, I think they can safely be ignored.

File: pkgs/os-specific/linux/lvm2/default.nix [hunk 2]
```diff
@@ -84,6 +85,7 @@ stdenv.mkDerivation rec {
       url = "https://git.alpinelinux.org/aports/plain/main/lvm2/mlockall-default-config.patch?h=3.7-stable&id=31bd4a8c2dc00ae79a821f6fe0ad2f23e1534f50";
       sha256 = "1ivbj3sphgf8n1ykfiv5rbw7s8dgnj5jcr9jl2v8cwf28lkacw5l";
     })
+    ./no-dynamic.patch
```
[the remark this answers is not in the document]
Comment (baloo) [reply pkgs/os-specific/linux/lvm2/default.nix:92]: https://www.redhat.com/archives/lvm-devel/2020-August/msg00016.html

**Review by 7c6f434c — APPROVED** [review]
Code looks fine to me

**Pull request closed without merging**